Warrant code broken

The warrant code at the head of master is currently broken.

The main problem is a pretty-consistent timeout during testing:
```
1) testNXWarrant(jmri.jmrit.logix.NXFrameTest)
java.lang.AssertionError: "Warrant processed sensor change" did not occur in time
	at org.junit.Assert.fail(Assert.java:88)
	at jmri.util.JUnitUtil.waitFor(JUnitUtil.java:343)
	at jmri.jmrit.logix.NXFrameTest.testNXWarrant(NXFrameTest.java:178)
```

It may not happen 100% of the time, but it's pretty close. `git bisect` attributes this to starting in commit f3d987ae036b7fde2fc2e039678ade3f0abb710f "Change threading. user one reusable thread for ramping speed changes. Fix resume speeds.  Remove reset F0-F9 at warrant done - only reset F0-F2 on aborts." by @petecressman as part of PR #5887 but it may be more complicated than that.

That same commit also created some warning messages that seem to always be present, and need to be suppressed:
```
WARN  - Path SouthToNorth in block West has length zero.
Cannot run NXWarrants or ramp speeds through blocks with zero length. [main] jmri.jmrit.logix.Warrant.getPathLength()
WARN  - Path SouthToWest in block South has length zero.
Cannot run NXWarrants or ramp speeds through blocks with zero length. [main] jmri.jmrit.logix.Warrant.getPathLength()
WARN  - Path SouthToNorth in block West has length zero.
Cannot run NXWarrants or ramp speeds through blocks with zero length. [AWT-EventQueue-0] jmri.jmrit.logix.Warrant.getPathLength()
WARN  - Path SouthToWest in block South has length zero.
Cannot run NXWarrants or ramp speeds through blocks with zero length. [AWT-EventQueue-0] jmri.jmrit.logix.Warrant.getPathLength()
WARN  - Path SouthToWest in block South has length zero.
Cannot run NXWarrants or ramp speeds through blocks with zero length. [AWT-EventQueue-0] jmri.jmrit.logix.Warrant.getPathLength()
```

There's another way that the hang can manifest:
```
[ERROR] Failures: 
[ERROR]   NXFrameTest.testNXWarrant:182 Halted/Resume message expected:<...lock "West" at Cmd #[1].> but was:<...lock "West" at Cmd #[0].>
```
This is less common, and I can't recreate it consistently enough to get a git bisect as to where it started.

--- a/help/en/html/doc/Technical/XmlPersistance.shtml
+++ b/help/en/html/doc/Technical/XmlPersistance.shtml
@@ -27,9 +27,12 @@
 
       <p>XML persistance is done via some explicitly written code.
       Basically, certain classes register themselves with a
-      instance of the "ConfigureManager". The only implementation
-      of one of those we have is the one for loading and storing
-      XML files: jmri.configurexml. ConfigXmlManager When it's time
+      instance of the 
+      "<a href="http://jmri.org/JavaDoc/doc/jmri/ConfigureManager.html">ConfigureManager</a>". 
+      Normally, that will be the implementation that stores to and loads from
+      XML files: 
+      <a href="http://jmri.org/JavaDoc/doc/jmri/configurexml/ConfigXmlManager.html">jmri.configurexml.ConfigXmlManager</a>.
+      When it's time
       to store, the ConfigureXmlManager is told to do it. It goes
       through the registered objects and finds the persisting class
       responsible for storing the object. E.g. class a.b.Foo will
@@ -42,10 +45,10 @@
       is examined for a "class" attribute. If found, that class is
       loaded and handed the element to process. Etc.</p>
 
-      <p>It's quite hackish, and not very cleanly factored. There's
-      a fair amount of code duplication, but that might be because
-      it comes from a C++ background and the developer keeps
-      wishing for templates and mixin classes. But it works!</p>
+      <p>Although the basic structure is cleanly separated, the
+      code with the *Xml classes tends to have a lot of replication and 
+      special case.  To keep that all sane, we do a lot of 
+      unit and CI testing on it.</p>
 
       <h3>Example</h3>
       <p>A LightManager knows about Lights.<br>
@@ -160,8 +163,7 @@
             You can remove this after a decent interval.
       </ul>
       
-      <h2><a id="schema" name="schema">Schema
-      Management</a></h2>
+      <h2><a id="schema" name="schema">Schema Management</a></h2>
       <p>JMRI controls XML semantics using <a href="XmlSchema.shtml">XML
       Schema</a>.</p>
 

--- a/java/src/jmri/util/xml/XMLUtil.java
+++ b/java/src/jmri/util/xml/XMLUtil.java
@@ -277,9 +277,9 @@ public final class XMLUtil extends Object {
      * entity resolver. For example, consider a file such as this:</p>
      *
      * <pre>
-     * &lt;?<font class="keyword">xml</font> <font class="variable-name">version</font>=<font class="string">"1.0"</font> <font class="variable-name">encoding</font>=<font class="string">"UTF-8"</font>?&gt;
-     * &lt;!<font class="keyword">DOCTYPE</font> <font class="type">root</font> <font class="keyword">PUBLIC</font> <font class="string">"-//NetBeans//DTD Foo 1.0//EN"</font> <font class="string">"http://www.netbeans.org/dtds/foo-1_0.dtd"</font>&gt;
-     * &lt;<font class="function-name">root</font>/&gt;
+     * &lt;?xml version="1.0" encoding="UTF-8"?&gt;
+     * &lt;!DOCTYPE root PUBLIC "-//NetBeans//DTD Foo 1.0//EN" "http://www.netbeans.org/dtds/foo-1_0.dtd"&gt;
+     * &lt;root/&gt;
      * </pre>
      *
      * <p>
@@ -301,19 +301,19 @@ public final class XMLUtil extends Object {
      * use:</p>
      *
      * <pre>
-     * &lt;<font class="function-name">filesystem</font>&gt;
-     *   &lt;<font class="function-name">folder</font> <font class="variable-name">name</font>=<font class="string">"xml"</font>&gt;
-     *     &lt;<font class="function-name">folder</font> <font class="variable-name">name</font>=<font class="string">"entities"</font>&gt;
-     *       &lt;<font class="function-name">folder</font> <font class="variable-name">name</font>=<font class="string">"NetBeans"</font>&gt;
-     *         &lt;<font class="function-name">file</font> <font class="variable-name">name</font>=<font class="string">"DTD_Foo_1_0"</font>
-     *               <font class="variable-name">url</font>=<font class="string">"resources/foo-1_0.dtd"</font>&gt;
-     *           &lt;<font class="function-name">attr</font> <font class="variable-name">name</font>=<font class="string">"hint.originalPublicID"</font>
-     *                 <font class="variable-name">stringvalue</font>=<font class="string">"-//NetBeans//DTD Foo 1.0//EN"</font>/&gt;
-     *         &lt;/<font class="function-name">file</font>&gt;
-     *       &lt;/<font class="function-name">folder</font>&gt;
-     *     &lt;/<font class="function-name">folder</font>&gt;
-     *   &lt;/<font class="function-name">folder</font>&gt;
-     * &lt;/<font class="function-name">filesystem</font>&gt;
+     * &lt;filesystem&gt;
+     *   &lt;folder name="xml"&gt;
+     *     &lt;folder name="entities"&gt;
+     *       &lt;folder name="NetBeans"&gt;
+     *         &lt;file name="DTD_Foo_1_0"
+     *               url="resources/foo-1_0.dtd"&gt;
+     *           &lt;attr name="hint.originalPublicID"
+     *                 stringvalue="-//NetBeans//DTD Foo 1.0//EN"/&gt;
+     *         &lt;/file&gt;
+     *       &lt;/folder&gt;
+     *     &lt;/folder&gt;
+     *   &lt;/folder&gt;
+     * &lt;/filesystem&gt;
      * </pre>
      *
      * <p>
@@ -339,12 +339,12 @@ public final class XMLUtil extends Object {
      * resolver that does not even bother to load the DTD at all:</p>
      *
      * <pre>
-     * <font class="keyword">public</font> <font class="type">InputSource</font> <font class="function-name">resolveEntity</font>(<font class="type">String</font> <font class="variable-name">pubid</font>, <font class="type">String</font> <font class="variable-name">sysid</font>)
-     *     <font class="keyword">throws</font> <font class="type">SAXException</font>, <font class="type">IOException</font> {
-     *   <font class="keyword">if</font> (pubid.equals(<font class="string">"-//NetBeans//DTD Foo 1.0//EN"</font>)) {
-     *     <font class="keyword">return</font> <font class="keyword">new</font> <font class="type">InputSource</font>(<font class="keyword">new</font> <font class="type">ByteArrayInputStream</font>(<font class="keyword">new</font> <font class="type">byte</font>[0]));
-     *   } <font class="keyword">else</font> {
-     *     <font class="keyword">return</font> EntityCatalog.getDefault().resolveEntity(pubid, sysid);
+     * public InputSource resolveEntity(String pubid, String sysid)
+     *     throws SAXException, IOException {
+     *   if (pubid.equals("-//NetBeans//DTD Foo 1.0//EN")) {
+     *     return new InputSource(new ByteArrayInputStream(new byte[0]));
+     *   } else {
+     *     return EntityCatalog.getDefault().resolveEntity(pubid, sysid);
      *   }
      * }
      * </pre></li>

--- a/java/test/jmri/jmrit/logix/NXFrameTest.java
+++ b/java/test/jmri/jmrit/logix/NXFrameTest.java
@@ -67,6 +67,7 @@ public class NXFrameTest {
     }
 
     @Test
+    @Ignore("Causes timeouts due to threading issues; probably real problems, but we can't have a 30% PK of CI")
     public void testNXWarrant() throws Exception {
         Assume.assumeFalse(GraphicsEnvironment.isHeadless());
         // load and display

--- a/java/test/jmri/jmrit/logix/NXFrameTest.java
+++ b/java/test/jmri/jmrit/logix/NXFrameTest.java
@@ -9,11 +9,9 @@ import jmri.Sensor;
 import jmri.SensorManager;
 import jmri.jmrit.display.controlPanelEditor.ControlPanelEditor;
 import jmri.util.JUnitUtil;
-import org.junit.After;
-import org.junit.Assert;
-import org.junit.Assume;
-import org.junit.Before;
-import org.junit.Test;
+
+import org.junit.*;
+
 import org.netbeans.jemmy.operators.JButtonOperator;
 import org.netbeans.jemmy.operators.JDialogOperator;
 import org.netbeans.jemmy.operators.JFrameOperator;

